src/Lecture3.hs

```diff
@@ -208,10 +232,28 @@ types that can have such an instance.
 -}
 
 -- instance Foldable Weekday where
+-- this instance does not make sense since Weekday it's an enumeration
+
 -- instance Foldable Gold where
+-- this instance does not make sense since it would be an implementation of Foldable for Int
+
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+-- this instance does not make sense since it would be an implementation of Foldable for Int
+
+instance Foldable List1 where
+    foldr :: (a -> b -> b) -> b -> List1 a -> b
+    foldr f z (List1 x xs) = f x $ foldr f z xs
+
+    foldMap :: Monoid m => (a -> m) -> List1 a -> m
+    foldMap f = foldr ((<>) . f) mempty
```

Comment:
That's already the default implementation of `foldMap` so there's not much benefit in implementing it explicitly in exactly the same way 🙂 
But maybe you can think of a different implementation 🤔